intellij-plugin-structure/structure-classes/src/main/java/com/jetbrains/plugin/structure/classes/utils/KtClassResolver.kt

```diff
@@ -0,0 +1,71 @@
+package com.jetbrains.plugin.structure.classes.utils
+
+import kotlinx.metadata.jvm.KotlinClassMetadata
+import kotlinx.metadata.jvm.Metadata
+import org.objectweb.asm.tree.AnnotationNode
+import org.objectweb.asm.tree.ClassNode
+import java.util.*
+
+internal const val KOTLIN_METADATA_ANNOTATION_DESC = "Lkotlin/Metadata;"
+
+class KtClassResolver {
+  private val cache = TreeMap<ClassNode, KtClassNode>(Comparator<ClassNode> { o1, o2 ->
```

Comment:
Why do you need to use `TreeMap` here, why isn't it enough to use a `HashMap`?

Reply:
`ClassNode` is from ASM library and it does not provide `hashCode()`/`equals()` which is required by `HashMap`.

Reply:
I meant that you can use `HashMap<String, KtClassNode>` and use `signature` as a key like it's done now.